From 7cbf2b33224c7a2b85954b90627fa0b5d12bbc89 Mon Sep 17 00:00:00 2001
From: Huazhong Tan <tanhuazhong@huawei.com>
Date: Tue, 2 Feb 2021 20:39:53 +0800
Subject: [PATCH 11/15] net: hns3: remove unnecessary check in hns3_dbg_read()

Clang warns:

drivers/net/ethernet/hisilicon/hns3/hns3_debugfs.c:499:28: warning:
address of array 'filp->f_path.dentry->d_iname' will always evaluate
to 'true' [-Wpointer-bool-conversion]
        if (!filp->f_path.dentry->d_iname)

This check is unnecessary, so remove it.

Reported-by: Nathan Chancellor <nathan@kernel.org>
Signed-off-by: Huazhong Tan <tanhuazhong@huawei.com>
Link: https://lore.kernel.org/r/1612269593-18691-8-git-send-email-tanhuazhong@huawei.com
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/net/ethernet/hisilicon/hns3/hns3_debugfs.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/net/ethernet/hisilicon/hns3/hns3_debugfs.c b/drivers/net/ethernet/hisilicon/hns3/hns3_debugfs.c
index 6978304f1ac5..c5958754f939 100644
--- a/drivers/net/ethernet/hisilicon/hns3/hns3_debugfs.c
+++ b/drivers/net/ethernet/hisilicon/hns3/hns3_debugfs.c
@@ -494,9 +494,6 @@ static ssize_t hns3_dbg_read(struct file *filp, char __user *buffer,
 	ssize_t size = 0;
 	int ret = 0;
 
-	if (!filp->f_path.dentry->d_iname)
-		return -EINVAL;
-
 	read_buf = kzalloc(HNS3_DBG_READ_LEN, GFP_KERNEL);
 	if (!read_buf)
 		return -ENOMEM;
-- 
2.30.0

